assets/src/components/mapPage.tsx

```diff
@@ -269,20 +275,35 @@ const MapPage = (): ReactElement<HTMLDivElement> => {
     [dispatch]
   )
 
-  const selectSearchResult = useCallback(
+  const selectVehicleResult = useCallback(
     (vehicleOrGhost: Vehicle | Ghost | null) => {
       if (vehicleOrGhost) {
-        setSelection({
+        setVehicleSelection({
           type: SelectedEntityType.Vehicle,
           vehicleId: vehicleOrGhost.id,
         })
       } else {
-        setSelection(null)
+        setVehicleSelection(null)
       }
     },
-    [setSelection]
+    [setVehicleSelection]
   )
 
+  const selectLocationResult = (
```

Comment:
question (non-blocking): Is there a benefit to introducing `setLocationResult` over using the `setSelection` function? I wonder if there may be an advantage to keeping it a single function and a single `selectSearchResult` prop for `SearchMode`. That way  if we add something else like Route to search results, we wouldn't have to add another selection prop to `SearchMode`.

Reply:
I definitely debated this. My thinking is that the existing function that I renamed `selectVehicleResult` already had vehicle-specific logic in it and having vehicles, locations, and potentially other kinds of results set by the same function would likely end up involving a lot of user-defined type guards to differentiate them, which I'm not a big fan of. That said, I see your point that as we add further search result types, this might actually end up being the more complicated approach. I'll think about it.